### PR TITLE
Finish Parameter Mappings for `net.minecraft.client.gui`

### PR DESCRIPTION
**Pull request by ChampionAsh5357**
Adds the rest of the missing parameters for `net.minecraft.client.gui`. Lambdas are not included since I don't remember whether those are allowed :p.

### CODE REVIEW

**Comment by alcatrazEscapee** [comment]
Lambdas are allowed, yes, and encouraged, because other people not using Forge exist and can benefit from said mappings.

File: data/net/minecraft/client/gui/screens/worldselection/CreateWorldScreen.mapping [hunk 1]
```diff
@@ -28,6 +28,7 @@ CLASS net/minecraft/client/gui/screens/worldselection/CreateWorldScreen
 		ARG 1 debug
 	METHOD createNewWorld (Lnet/minecraft/world/level/storage/PrimaryLevelData$SpecialWorldProperty;Lnet/minecraft/core/LayeredRegistryAccess;Lcom/mojang/serialization/Lifecycle;)V
 		ARG 1 specialWorldProperty
+		ARG 2 registries
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/gui/screens/worldselection/CreateWorldScreen.mapping]:
```suggestion
		ARG 2 registryAccess
```

File: data/net/minecraft/client/gui/screens/recipebook/OverlayRecipeComponent.mapping [hunk 2]
```diff
@@ -4,6 +4,9 @@ CLASS net/minecraft/client/gui/screens/recipebook/OverlayRecipeComponent
 		ARG 2 collection
 		ARG 3 x
 		ARG 4 y
+		ARG 5 copmonentXCenter
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/gui/screens/recipebook/OverlayRecipeComponent.mapping]:
```suggestion
		ARG 5 componentXCenter
```

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]

**Comment by sciwhiz12** [comment]
Due to the length of time and versions past since this was opened, I'm going to close this PR as stale. If you want to continue this PR, be my guest and reopen the PR. Thanks!

**Pull request closed without merging**